imi/src/test/java/org/motechproject/nms/imi/it/CdrFileServiceBundleIT.java

```diff
@@ -89,18 +111,76 @@ public void testServicePresent() {
 
 
     @Test
-    public void testParse() throws IOException, NoSuchAlgorithmException {
-        getLogger().debug("testParse()");
+    public void testVerify() throws IOException, NoSuchAlgorithmException {
+        getLogger().debug("testVerify()");
+
+        CdrHelper helper = new CdrHelper(settingsService, subscriptionService, subscriberDataService,
+                languageDataService, languageLocationDataService, circleDataService, stateDataService,
+                districtDataService, fileAuditRecordDataService);
+
+        helper.makeCdrs(1,1,1,1);
+        helper.makeCdrFile();
+        FileInfo fileInfo = new FileInfo(helper.cdr(), helper.cdrChecksum(), helper.cdrCount());
+        Map<String, Object> eventParams = new HashMap<>();
+        VerifyResults result = cdrFileService.verifyDetailFile(fileInfo);
+        assertEquals(0, result.getErrors().size());
+    }
+
+
+    @Test(expected = IllegalStateException.class)
```

Comment:
Instead of using expected for the entire test I've been defining a rule so I can verify the exception is thrown on the line I expect it to be thrown from.

http://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests/2935935#2935935